extensions/stackdriver/log/exporter.cc

```diff
@@ -42,19 +42,23 @@ namespace Stackdriver {
 namespace Log {
 
 ExporterImpl::ExporterImpl(RootContext* root_context,
-                           const std::string& logging_service_endpoint)
-    : export_call_(MetricType::Counter, "stackdriver_logging_filter",
-                   {MetricTag{"type", MetricTag::TagType::String},
-                    MetricTag{"success", MetricTag::TagType::Bool}}) {
+                           const std::string& logging_service_endpoint) {
   context_ = root_context;
-  success_callback_ = [this](google::protobuf::Empty&&) {
-    export_call_.increment(1, "logging", true);
+  Metric export_call(MetricType::Counter, "stackdriver_filter",
+                     {MetricTag{"type", MetricTag::TagType::String},
+                      MetricTag{"success", MetricTag::TagType::Bool}});
+  auto success_counter = export_call.resolve("logging", true);
+  auto failure_counter = export_call.resolve("logging", false);
+  success_callback_ = [success_counter](google::protobuf::Empty&&) {
+    // TODO(bianpengyuan): replace this with envoy's generic gRPC counter.
+    incrementMetric(success_counter, 1);
```

Comment:
@jplevyak this was our solution to simplify the callback by not capturing references.
This lambda just captures the returned `int` and will use the C API to call into the host.
What is the behaviour if `incrementMetric` is called with an invalid id ?

Reply:
The wasm container stores a map from int -> metric and does not free any metrics during the lifetime of the VM, so a valid int for a metric will remain so as long as the VM lives.